### PR TITLE
Update query fails if offset is given

### PR DESCRIPTION
**Pull request by Deltachaos**
.

### CODE REVIEW

**Comment by brianteeman** [comment]
Please provide step by step instrcutions how to replicate the error and
test the proposed solution

On 4 June 2015 at 12:05, Maximilian Ruta <notifications@github.com> wrote:

> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/joomla/joomla-cms/pull/7119
> Commit Summary
>
>    - Update query fails if offset is given
>
> File Changes
>
>    - *M* libraries/joomla/database/query/mysqli.php
>    <https://github.com/joomla/joomla-cms/pull/7119/files#diff-0> (9)
>
> Patch Links:
>
>    - https://github.com/joomla/joomla-cms/pull/7119.patch
>    - https://github.com/joomla/joomla-cms/pull/7119.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/7119>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by brianteeman** [comment]
After over one month without further reply to the request for further information this issue is being closed. It can always be re-opened
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7119">issues.joomla.org/joomla-cms/7119</a>.</sub>

**Comment by nibra** [comment]
The method produces invalid SQL, if $limit is empty (eg, $limit == '').
The proposed code fixes that.

File: libraries/joomla/database/query/mysqli.php [hunk 1]
```diff
@@ -45,7 +45,14 @@ public function processLimit($query, $limit, $offset = 0)
 	{
 		if ($limit > 0 || $offset > 0)
 		{
```
Comment (nibra) [inline libraries/joomla/database/query/mysqli.php]: $limit and $offset needs to be cast to integer (this is a publicly available method, so you never know)

File: libraries/joomla/database/query/mysqli.php [hunk 2]
```diff
@@ -45,7 +45,14 @@ public function processLimit($query, $limit, $offset = 0)
...
-			$query .= ' LIMIT ' . $offset . ', ' . $limit;
+			if ($offset > 0)
+			{
+				$query .= ' LIMIT ' . $offset . ', ' . $limit;
```
Comment (nibra) [inline libraries/joomla/database/query/mysqli.php]: Fails, if $limit is an empty string...

**Pull request closed without merging**